### PR TITLE
Add Comment API Module

### LINKED ISSUE #11: Add Comments API module
Follow [Basecamp3 API comments section](https://github.com/basecamp/bc3-api/blob/master/sections/comments.md)

The API module should cover the following endpoints:
 - [Add a comment](https://github.com/basecamp/bc3-api/blob/master/sections/comments.md#create-a-comment)
 - [Get comments](https://github.com/basecamp/bc3-api/blob/master/sections/comments.md#get-comments)

### PR DESCRIPTION
**Pull request by jcurbelo**
## Description

Fix #11 

Adds `CommentAPI` module to fetch and create comments for any given `Resource`.

### Example Usage

```ruby
require 'camp3'

client = Camp3.configure do |config|
  config.client_id = ENV['BASECAMP3_CLIENT_ID']
  config.client_secret = ENV['BASECAMP3_CLIENT_SECRET']
  config.account_number = ENV['BASECAMP3_ACCOUNT_NUMBER']
  config.refresh_token = ENV['BASECAMP3_REFRESH_TOKEN']
  config.access_token = ENV['BASECAMP3_ACCESS_TOKEN']
end

# Resource from URL
resource = client.resource('YOUR_RESOURCE_URL')
if resource.can_be_commented?
  client.add_comment(resouce, 'Hello camp3, nice project!')
  comments = client.comments(resouce)
  comments.each do |c|
    puts "#{c.content}"
  end
end
```

For more examples check **./examples/comments.rb*

## Changes

 * Added method `can_be_commented?` to check whether a resource can be commented or not
 * Fix: Added `'Content-Type' => 'application/json'` request header and updating its values before doing any request
 * Fix: Initializing default `user_agent` with `DEFAULT_USER_AGENT` in `Configuration.new` constructor
 * Added `CommentAPI` module with `comments` and `add_comment` methods

### CODE REVIEW

File: lib/camp3/request.rb [hunk 1]
```diff
@@ -67,6 +67,7 @@ def self.decode(response)
 
     # Executes the request
     def execute_request(method, endpoint, params)
+      params[:headers].merge!(self.class.headers)
```
Comment (jcurbelo) [inline lib/camp3/request.rb:70]: Passing default headers along with the authz headers

File: lib/camp3/request.rb [hunk 2]
```diff
@@ -8,7 +8,7 @@ class Request
     include HTTParty
     include Logging
     format :json
-    headers 'Accept' => 'application/json'
+    headers 'Accept' => 'application/json', 'Content-Type' => 'application/json'
```
Comment (jcurbelo) [inline lib/camp3/request.rb:11]: Assumed that `format :json` was already doing this, but for some reason, a `400` response is returned if the explicit **Content-Type** header is missing

File: lib/camp3/configuration.rb [hunk 3]
```diff
@@ -23,6 +23,7 @@ class Configuration
     attr_accessor(*VALID_OPTIONS_KEYS)
 
     def initialize(options = {})
+      options[:user_agent] ||= DEFAULT_USER_AGENT
```
Comment (jcurbelo) [inline lib/camp3/configuration.rb:26]: Safety check

File: lib/camp3/request.rb (hunk 2 above)
Comment (renehernandez) [reply lib/camp3/request.rb:11]: According to [making a request](https://github.com/basecamp/bc3-api#making-a-request), **Content-Type** is required for requests that create new resources, like `add_comments`. That's a great finding.

**Pull request merged**